src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -427,56 +433,75 @@ def run_consent_request(
         policy: Policy,
         privacy_request: PrivacyRequest,
         identity_data: Dict[str, Any],
-        consent_preferences: List[Consent],
+        consent_preferences: List[PrivacyRequestConsentPreference],
     ) -> bool:
         """Execute a consent request. Return whether the consent request to the third party succeeded.
-        Return True if 200 OK
+        Should only propagate either the entire set of opt in or opt out requests.
+        Return True if 200 OK. Raises a SkippingConsentPropagation exception if no action is taken
+        against the service.
         """
         logger.info(
             "Starting consent request for node: '{}'",
             node.address.value,
         )
         self.set_privacy_request_state(privacy_request, node)
         query_config = self.query_config(node)
+        should_opt_in: Optional[bool] = should_opt_in_to_service(
```

Comment:
This method has all the new logic as to whether or not we're going to send the set of opt in or opt out requests to the third party.  Subject to change as requirements solidify.